src/SFA.DAS.CandidateAccount.Application.UnitTests/WorkExperience/WhenHandlingGetApplicationWorkHistoriesQueryHandler.cs

```diff
@@ -0,0 +1,28 @@
+﻿using AutoFixture.NUnit3;
+using FluentAssertions;
+using Moq;
+using SFA.DAS.CandidateAccount.Application.Application.Queries.GetApplicationWorkHistories;
+using SFA.DAS.CandidateAccount.Data.WorkExperience;
+using SFA.DAS.CandidateAccount.Domain.Application;
+using SFA.DAS.Testing.AutoFixture;
+
+namespace SFA.DAS.CandidateAccount.Application.UnitTests.WorkExperience;
+
+
+[TestFixture]
+public class WhenHandlingGetApplicationWorkHistoriesQueryHandler
+{
+    [Test, RecursiveMoqAutoData]
+    public async Task Then_Request_Is_Handled_And_Entities_Returned(
+        GetApplicationWorkHistoriesQuery request,
+        List<WorkHistoryEntity> entities,
+        [Frozen] Mock<IWorkHistoryRepository> workExperienceRepository,
+        GetApplicationWorkHistoriesQueryHandler handler)
+    {
+        workExperienceRepository.Setup(x => x.Get(request.ApplicationId, CancellationToken.None)).ReturnsAsync(entities);
```

Comment:
Can we change this so that the CandidateId is also passed through to the repository call and filtered on as well? Currently it will allow me to get an application unrelated to a candidate I think 🤔

Reply:
Passing the candidate Id to the repo and filtering it based on Candidate ID